### PR TITLE
feat: Add `data_tiering_enabled`

### PR DESCRIPTION
**Pull request by Cheezmeister**
## what
* Introduce `var.data_tiering_enabled` and plumb it through to the aws provider

## why
* @teikametrics needs to set `data_tiering_enabled` true to provision our desired node type
  ```
  Error: error creating ElastiCache Replication Group (bidder-redis-cluster-production): InvalidParameterCombination: When     using the cache.r6gd.xlarge node type, you must enable data tiering.
  status code: 400, request id: REDACTED
  ```
* This flag `data_tiering_enabled` doesn't currently exist in the `terraform-aws-elasticache-redis` module

## references
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/elasticache_replication_group#data_tiering_enabled

### CODE REVIEW

**Comment by Cheezmeister** [comment]
The build fails with message

```
+ echo 'Changes detected. Pushing to the PR branch'
```

Please, what are the changes, and what access is required to complete CI?

File: variables.tf [hunk 1]
```diff
@@ -112,6 +112,12 @@ variable "apply_immediately" {
   description = "Apply changes immediately"
 }
 
+variable "data_tiering_enabled" {
+  type        = bool
+  default     = false
+  description = "Enables data tiering. Data tiering is only supported for replication groups using the r6gd node type. See: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/elasticache_replication_group#data_tiering_enabled"
```
Comment (nitrocode) [inline variables.tf:118]: You already copied this information from the link so I do not think we need it anymore.

```suggestion
  description = "Enables data tiering. Data tiering is only supported for replication groups using the r6gd node type."
```

Comment (nitrocode) [reply variables.tf:118]: After this change, you'll have to run `make pr/auto-format` again

**Review by nitrocode — DISMISSED** [review]

**Review by nitrocode — APPROVED** [review]

**Pull request merged**